packages/app-project/src/components/ProjectHeader/ProjectHeader.js

```diff
@@ -1,29 +1,37 @@
 import { Box } from 'grommet'
-import { array, arrayOf, bool, shape, string } from 'prop-types'
+import { observer } from 'mobx-react'
+import { bool, string } from 'prop-types'
 import styled from 'styled-components'
-import { withResizeDetector } from 'react-resize-detector'
+import { useResizeDetector } from 'react-resize-detector'
 
-import ApprovedIcon from './components/ApprovedIcon'
-import Avatar from './components/Avatar'
-import Background from './components/Background'
-import DropdownNav from './components/DropdownNav'
-import LocaleSwitcher from './components/LocaleSwitcher'
-import Nav from './components/Nav'
-import ProjectTitle from './components/ProjectTitle'
-import UnderReviewLabel from './components/UnderReviewLabel'
+import {
+  ApprovedIcon,
+  Avatar,
+  Background,
+  DropdownNav,
+  LocaleSwitcher,
+  Nav,
+  ProjectTitle,
+  UnderReviewLabel
+} from './components/'
+import { useProjectNavigation, useStores } from './hooks/'
 
 const StyledBox = styled(Box)`
   position: relative;
 `
 
 function ProjectHeader({
-  availableLocales = [],
-  className = '',
-  inBeta = false,
-  navLinks = [],
-  title,
-  width
+  adminMode,
+  className = ''
 }) {
+  const { width, height, ref } = useResizeDetector()
+  const navLinks = useProjectNavigation(adminMode)
+  const {
+    availableLocales,
```

Comment:
This could be moved down into the `LocaleSwitcher` component. I don’t think it’s used anywhere in the project header.

Reply:
On second thoughts, `ProjectHeader` is using `availableLocales` to calculate layout. Moving it would break those calculations.